src/Lean/Server/References.lean

```diff
@@ -210,6 +210,24 @@ def referringTo (self : References) (ident : RefIdent) (srcSearchPath : SearchPa
           result := result.push ⟨uri, range⟩
   result
 
+def definitionsMatching (self : References) (srcSearchPath : SearchPath) (filter : Name -> Bool)
+    (maxAmount : Option Nat := none) : IO $ Array (Name × Location) := do
+  let mut result := #[]
+  let mut amount := 0
+  for (module, refs) in self.allRefs.toList do
+    if let some path ← srcSearchPath.findModuleWithExt "lean" module then
+      let uri := DocumentUri.ofPath <| ← IO.FS.realPath path
```

Comment:
Doing hundreds of IO ops no matter the result size seems a bit wasteful, though perhaps we should first measure if it matters compared to the filtering overhead.